lib/utils.ex

```diff
@@ -0,0 +1,27 @@
+defmodule HashRing.Utils do
+  @compile :native
+
+  @type t :: __MODULE__
+
+  def hash(key) when is_binary(key) do
+    <<_ :: binary-size(8), value :: unsigned-little-integer-size(64)>> = :erlang.md5(key)
+    value
+  end
+  def hash(key), do: hash("#{key}")
+
+  def gen_items([], _num_replicas), do: {}
+  def gen_items(nodes, num_replicas) do
+    gen_items(nodes, Enum.to_list(0..(num_replicas - 1)), [])
+  end
+
+  def gen_items([], _replicas, items) do
+    items
+      |> Enum.sort(&(elem(&1, 0) < elem(&2, 0)))
+      |> List.to_tuple
```

Comment:
This should be removed from here and used directly in the other hash ring. It is a waste to perform this operation.